Makefile

```diff
@@ -100,3 +92,99 @@ tox:
+
+# Set EDITABLE to -e to install the package in editable mode outside of CI. This is
+# useful for local development.
+ifdef CI
+EDITABLE=
+else
+EDITABLE=-e
+endif
+
+# Date until which dependencies installed with --exclude-newer must have been released.
+# Dependencies released after this date are ignored.
+EXCLUDE_NEWER_DATE="2024-08-07"
+
+# Install package for local development.
+.PHONY: install-dev
+install-dev:
+	uv pip install ${EDITABLE} . --all-extras --requirement pyproject.toml
+
+
+# Install package with API dependencies only.
+# Should be same command as in the docs with some extra flags for CI:
+# https://docs.lightly.ai/docs/install-lightly#install-the-lightly-python-client
+.PHONY: install-api-only
+install-api-only:
+	uv pip install --exclude-newer ${EXCLUDE_NEWER_DATE} --requirement requirements/base.txt
+	uv pip install --exclude-newer ${EXCLUDE_NEWER_DATE} . --no-deps
+
+# Install package with minimal dependencies.
+# 
+# This command is split into multiple steps:
+# 1. Install the dev dependencies to be able to run tests. We don't want to use
+#    the minimal versions for these dependencies.
+# 2. Then we reinstall the package with minimal dependencies.
+# 3. Finally we install setuptools<50. This is necessary for compatibility with old
+#    PyTorch Lightning versions that do not include the correct setuptools dependencies.
+#
+# Explanation of flags:
+# --exclude-newer: We don't want to install dependencies released after that date to
+#   keep CI stable.
+# --resolution=lowest-direct: Only install minimal versions for direct dependencies.
+#   Transitive dependencies will use the latest compatible version.
+# 	Using --resolution=lowest would also download the latest versions for transitive
+#   dependencies which is not a realistic scenario and results in some extremely old
+#   dependencies being installed.
+# --reinstall: Reinstall dependencies to make sure they satisfy the constraints.
+.PHONY: install-minimal
+install-minimal:
+	uv pip install --exclude-newer ${EXCLUDE_NEWER_DATE} ${EDITABLE} ".[dev]"
+	uv pip install --resolution=lowest-direct --exclude-newer ${EXCLUDE_NEWER_DATE} --reinstall ${EDITABLE} ".[minimal]"
+	uv pip install --exclude-newer ${EXCLUDE_NEWER_DATE} --reinstall "setuptools<50"
+
+# Install package with minimal dependencies including extras.
+# See install-minimal for explanation of flags.
+# We do not use --all-extras because it includes the dev dependencies for which we don't
+# want to install the minimal versions.
+.PHONY: install-minimal-extras
+install-minimal-extras:
+	uv pip install --exclude-newer ${EXCLUDE_NEWER_DATE} ${EDITABLE} ".[dev]"
+	uv pip install --resolution=lowest-direct --exclude-newer ${EXCLUDE_NEWER_DATE} --reinstall ${EDITABLE} ".[matplotlib,minimal,timm,video]" --requirement pyproject.toml
+	uv pip install --exclude-newer ${EXCLUDE_NEWER_DATE} --reinstall "setuptools<50"
+
+# Install package with fixed dependencies.
+.PHONY: install-fixed
+install-fixed:
```

Comment:
I think I said this earlier to Malte. The name. `fixed` is not descriptive enough. What are the dependencies fixed to? And why?

Reply:
I changed it to `install-pinned` and described it in more detail in the comment.